test/test_selector.js

```diff
@@ -189,4 +189,40 @@ suite('selector', () => {
         assert.equal(memoized('A'), 'A');
         assert.equal(called, 2);
     });
+    test("composing selectors", function() {
+        const selector = composeSelectors({
+              x: state => state.a,
+              y: state => state.b
+        });
+
+        let firstResult = selector({a: 1,b: 2});
+        assert.deepEqual( firstResult, {x: 1, y: 2});
+        assert.strictEqual(selector({a: 1,b: 2}), firstResult);
+        assert.notStrictEqual(selector({a: 2,b: 2}),firstResult);
+        assert.notEqual(selector({a: 2,b: 2}), firstResult);
+        assert.deepEqual(selector({a: 2,b: 2}), {x: 2,y: 2});
+    });
+    test("composing selectors with custom selector creator", function() {
+        let customEqualityFunctionCalls = 0;
+        const customSelectorCreator = createSelectorCreator(
+            defaultMemoize,
+            (a,b) => {
+                ++customEqualityFunctionCalls;
+                return JSON.stringify(a) == JSON.stringify(b)
```

Comment:
The stringify is simply to demonstrate that any custom equality function may be used.
The counting only ensures, that the equality function is not being called to often while checking the arguments for equality. But with the other tests still passing, this is no necessity either.
I am gonna simplify the case.